### PR TITLE
Switched from ip to tz for geolocation

### PR DESCRIPTION
**Pull request by samhotep**
## Done

- Switched from using ip address for geolocation to using timezone

## QA

- Open https://ubuntu-com-13636.demos.haus/contact-us/form?product=generic-contact-us 
  - Check that the phone number input shows the correct country flag for your location.  
  - Be sure to test on mobile, tablet and desktop screen sizes
  
## Issue / Card

Fixes [WD-1173](https://warthogs.atlassian.net/browse/WD-1173)

## Screenshots

image

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

## Follow up
Should we also auto-fill the country input? 

[WD-1173]: https://warthogs.atlassian.net/browse/WD-1173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/views.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # Standard library
+import datetime
+import html
```
Comment (carkod) [inline webapp/views.py:3]: This doesn't seem to be used anywhere?

Comment (samhotep) [reply webapp/views.py:3]: Removing!

Comment (carkod) [reply webapp/views.py]: Sorry, it's not used by your code, but it seems we are using it here https://github.com/canonical/ubuntu.com/pull/13636/files#diff-eb0c47c22a26e963ac57d9840cff927d3d40d2f0889900c17e6ed717459a2db6L911

This file is becoming too big...

**Review by carkod — CHANGES_REQUESTED** [review]
This works for your example with telephone numbers, but because /user-country.json needs the query arg, this is breaking takeovers geolocation https://github.com/canonical/ubuntu.com/blob/8ac389f3332ed038af3a91173ee5990835534b7d/templates/base_index.html#L1417

You may need to do the `Intl.DateTimeFormat().resolvedOptions().timeZone;` for takeovers too

File: webapp/app.py [hunk 2]
```diff
@@ -628,7 +628,7 @@ def takeovers_index():
 core_services_guide.init_app(app)
 
 
-app.add_url_rule("/user-country.json", view_func=get_user_country_by_ip)
+app.add_url_rule("/user-country.json", view_func=get_user_country_by_tz)
```
Comment (carkod) [inline webapp/app.py:631]: If you are replacing this completely, then you might as well delete `get_user_country_by_ip` function

File: webapp/views.py [hunk 3]
```diff
@@ -1071,6 +1072,27 @@ def thank_you():
     )
 
 
+def get_user_country_by_tz():
```
Comment (carkod) [inline webapp/views.py:1075]: Can you also write a comment here for this function explaining how it works? I guess the key of this system is to use `Intl.DateTimeFormat().resolvedOptions().timeZone;`

**Comment by carkod** [comment]
Also can you remove his `Keyerror`, this was put there to check which IPs are not detected, which is not needed anymore with this PR.

File: webapp/app.py (hunk 2 above)
Comment (samhotep) [reply webapp/app.py]: Ok, but will do separately when updating the takeovers as well

**Review by carkod — APPROVED** [review]
LGTM, thanks.

**Pull request merged**